src/qt/psbtoperationsdialog.cpp

```diff
@@ -0,0 +1,255 @@
+        filename_suggestion.append(address_str + "-" + amount);
+        first = false;
+    }
+    filename_suggestion.append(".psbt");
+    QString filename = GUIUtil::getSaveFileName(this,
+        tr("Save Transaction Data"), filename_suggestion,
+        tr("Partially Signed Transaction (Binary) (*.psbt)"), &selected_filter);
+    if (filename.isEmpty()) {
+        return;
+    }
+    std::ofstream out(filename.toLocal8Bit().data());
+    out << ssTx.str();
+    out.close();
+    showStatus(tr("PSBT saved to disk."), StatusLevel::INFO);
+}
+
+void PSBTOperationsDialog::updateTransactionDisplay() {
+    m_ui->transactionDescription->setText(QString::fromStdString(renderTransaction(m_transaction_data)));
+    showTransactionStatus(m_transaction_data);
+}
+
+std::string PSBTOperationsDialog::renderTransaction(const PartiallySignedTransaction &psbtx)
+{
+    QString tx_description = "";
+    CAmount totalAmount = 0;
+    for (const CTxOut& out : psbtx.tx->vout) {
+        CTxDestination address;
+        ExtractDestination(out.scriptPubKey, address);
+        totalAmount += out.nValue;
+        tx_description.append(tr(" * Sends %1 to %2")
+            .arg(BitcoinUnits::formatWithUnit(BitcoinUnits::BTC, out.nValue))
+            .arg(QString::fromStdString(EncodeDestination(address))));
+        tx_description.append("<br>");
+    }
+
+    PSBTAnalysis analysis = AnalyzePSBT(psbtx);
+    tx_description.append(" * ");
+    if (!*analysis.fee) {
+        // This happens if the transaction is missing input UTXO information.
+        tx_description.append(tr("Unable to calculate transaction fee or total transaction amount."));
+    } else {
+        tx_description.append(tr("Pays transaction fee: "));
+        tx_description.append(BitcoinUnits::formatWithUnit(BitcoinUnits::BTC, *analysis.fee));
+
+        // add total amount in all subdivision units
+        tx_description.append("<hr />");
+        QStringList alternativeUnits;
+        for (const BitcoinUnits::Unit u : BitcoinUnits::availableUnits())
+        {
+            if(u != m_client_model->getOptionsModel()->getDisplayUnit())
+                alternativeUnits.append(BitcoinUnits::formatHtmlWithUnit(u, totalAmount));
+        }
+        tx_description.append(QString("<b>%1</b>: <b>%2</b>").arg(tr("Total Amount"))
+            .arg(BitcoinUnits::formatHtmlWithUnit(m_client_model->getOptionsModel()->getDisplayUnit(), totalAmount)));
+        tx_description.append(QString("<br /><span style='font-size:10pt; font-weight:normal;'>(=%1)</span>")
+            .arg(alternativeUnits.join(" " + tr("or") + " ")));
+    }
+
+    size_t num_unsigned = CountPSBTUnsignedInputs(psbtx);
+    if (num_unsigned > 0) {
+        tx_description.append("<br><br>");
+        tx_description.append(tr("Transaction has %1 unsigned inputs.").arg(QString::number(num_unsigned)));
+    }
+
+    return tx_description.toStdString();
+}
+
+void PSBTOperationsDialog::showStatus(const QString &msg, StatusLevel level) {
```

Comment:
Should we create a `OperationsDialog` class to make this stuff reusable? (can wait of course)

Reply:
I'm happy to go either way, are there other existing dialogs we'd immediately want to adopt it for? I do like it better than messageboxes, I tried to make it as drop-in as possible a replacement for them.